darts/explainability/tft_explainer.py

```diff
@@ -0,0 +1,249 @@
+        model
+            The fitted TFT model to be explained.
+        """
+        super().__init__(model)
+
+        if not model._fit_called:
+            raise ValueError("The model needs to be trained before explaining it.")
+
+        self._model = model
+
+    @property
+    def encoder_importance(self):
+        return self._get_importance(
+            weight=self._model.model._encoder_sparse_weights,
+            names=self._model.model.encoder_variables,
+        )
+
+    @property
+    def decoder_importance(self):
+        return self._get_importance(
+            weight=self._model.model._decoder_sparse_weights,
+            names=self._model.model.decoder_variables,
+        )
+
+    def get_variable_selection_weight(self, plot=False) -> Dict[str, pd.DataFrame]:
+        """Returns the variable selection weight of the TFT model.
+
+        Parameters
+        ----------
+        plot
+            Whether to plot the variable selection weight.
+
+        Returns
+        -------
+        TimeSeries
+            The variable selection weight.
+
+        """
+
+        if plot:
+            # plot the encoder and decoder weights
+            self._plot_cov_selection(
+                self.encoder_importance,
+                title="Encoder variable importance",
+            )
+            self._plot_cov_selection(
+                self.decoder_importance,
+                title="Decoder variable importance",
+            )
+
+        return {
+            "encoder_importance": self.encoder_importance,
+            "decoder_importance": self.decoder_importance,
+        }
+
+    def explain(self, **kwargs) -> ExplainabilityResult:
+        """Returns the explainability result of the TFT model.
+
+        The explainability result contains the attention heads of the TFT model.
+        The attention heads determine the contribution of time-varying inputs.
+
+        Parameters
+        ----------
+        kwargs
+            Arguments passed to the `predict` method of the TFT model.
+
+        Returns
+        -------
+        ExplainabilityResult
+            The explainability result containing the attention heads.
+
+        """
+        super().explain()
+        # without the predict call, the weights will still bet set to the last iteration of the forward() method
+        # of the _TFTModule class
+        if "n" not in kwargs:
+            kwargs["n"] = self._model.model.output_chunk_length
+
+        _ = self._model.predict(**kwargs)
+
+        # get the weights and the attention head from the trained model for the prediction
+        attention_heads = (
+            self._model.model._attn_out_weights.squeeze().sum(axis=1).detach()
+        )
+
+        # return the explainer result to be used in other methods
+        return ExplainabilityResult(
+            {
+                0: {
+                    "attention_heads": TimeSeries.from_dataframe(
+                        pd.DataFrame(attention_heads).transpose()
```

Comment:
If `attention_heads` is a NumPy array you could probably also do something simpler such as `TimeSeries.from_values(attention_heads.T)`.

Reply:
Thank you for the suggestion. I have changed to code accordingly